src/io/avro/write/schema.rs

```diff
@@ -48,13 +54,13 @@ fn _type_to_schema(data_type: &DataType) -> Result<AvroSchema> {
         DataType::Utf8 => AvroSchema::String(None),
         DataType::LargeUtf8 => AvroSchema::String(None),
         DataType::LargeList(inner) | DataType::List(inner) => AvroSchema::Array(Box::new(
-            type_to_schema(&inner.data_type, inner.is_nullable)?,
+            type_to_schema(&inner.data_type, inner.is_nullable, name_counter)?,
         )),
         DataType::Struct(fields) => AvroSchema::Record(Record::new(
-            "",
+            _get_field_name(name_counter),
             fields
                 .iter()
-                .map(field_to_field)
+                .map(|f| field_to_field(&f, name_counter))
```

Comment:
an alternative approach is to make the items this iteration as `struct_name.item1`, `struct_name.item2`, etc, but I imagine that `struct_name` would still need to be unique across the schema, right?

Reply:
Yes the names have to be globally unique